### PR TITLE
 Better loading of Button

### LINKED ISSUE #4925: Make Button Better
#### Environment(required)

- antd version: 2.7.1

Hey guys, can Button component add "delay" API？ like [delay](https://ant.design/components/spin/#components-spin-demo-delayAndDebounce) in Spin. 

I like this attribute, Sometimes, it make UI friendly and I can realize easily

### PR DESCRIPTION
**Pull request by RaoHai**
+ close #4925
 + add a default delay timer (300ms, same as spin)

### CODE REVIEW

File: components/button/button.tsx [hunk 1]
```diff
@@ -67,6 +68,30 @@ export default class Button extends React.Component<ButtonProps, any> {
...
+    if (currentLoading) {
+      clearTimeout(this.delayTimeout);
+    }
+
+    if (loading) {
+      this.delayTimeout = setTimeout(() => this.setState({ loading }), 300);
+    } else {
+      this.setState({ loading });
+    }
+
```
Comment (afc163) [inline components/button/button.tsx]: extra blank line.

File: components/button/demo/loading.md [hunk 2]
```diff
@@ -46,6 +56,9 @@ class App extends React.Component {
         <Button type="primary" icon="poweroff" loading={this.state.iconLoading} onClick={this.enterIconLoading}>
           Click me!
         </Button>
+        <Button type="primary" loading={this.state.delayLoading} onClick={this.delayLoading}>
+          Delay loading!
```
Comment (afc163) [inline components/button/demo/loading.md]: Delay loading! => Won't show loading

File: components/button/button.tsx [hunk 3]
```diff
@@ -67,6 +68,30 @@ export default class Button extends React.Component<ButtonProps, any> {
...
+  componentWillReceiveProps(nextProps) {
+    const currentLoading = this.props.loading;
+    const loading = nextProps.loading;
+
+    if (currentLoading) {
+      clearTimeout(this.delayTimeout);
+    }
+
+    if (loading) {
+      this.delayTimeout = setTimeout(() => this.setState({ loading }), 300);
```
Comment (afc163) [inline components/button/button.tsx]: 300 => 200

300 is too slow.

**Comment by RaoHai** [comment]
done

**Pull request merged**